src/checkpoints/checkpoints.cpp

```diff
@@ -166,6 +166,29 @@ namespace cryptonote
         block.major_version < network_version_12_checkpointing)
       return;
 
+    if (m_nettype == MAINNET && height == HF_VERSION_12_CHECKPOINTING_SOFT_FORK_HEIGHT)
+    {
+      uint64_t start_height = 0;
+      get_newest_hardcoded_checkpoint(m_nettype, &start_height);
+      start_height += 1; // Don't start deleting from the hardcoded height
+
+      for (uint64_t delete_height = start_height + (start_height % service_nodes::CHECKPOINT_INTERVAL);
```

Comment:
This start height doesn't look right (e.g. `start_height=1003` will start at `delete_height=1006` then 1010, etc., but it should presumably go through 1004, 1008, 1012, ....

Reply:
Fixed